[BUG] TavilyWebSearchEngine - Illegal character in path at index ~

<!--
Please note that the maintainer of this project is on vacation until the 5th of August.
During this period, there might be delays in responses to issues and pull requests.
Thank you for your understanding and patience.
-->

<!-- Please provide as many details as possible, this will help us to deliver a fix as soon as possible. Thank you! -->

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
URI.create cause Illegal charater exception.
After exception, all langchain4j pipeline stop to work. very critical and unsafe to use.

TavilyWebSearch result with space included url always reproduce this bug.


**Log and Stack trace**
<!-- Please provide a log and a stack trace (with exception), if applicable. -->
```java
java.lang.IllegalArgumentException: Illegal character in path at index 25: https://cboard.net/s/너희들이 게맛을 알어
        at java.base/java.net.URI.create(URI.java:932) ~[na:na]
        at dev.langchain4j.web.search.tavily.TavilyWebSearchEngine.toWebSearchOrganicResult(TavilyWebSearchEngine.java:101) ~[langchain4j-web-search-engine-tavily-0.33.0.jar:na]
        
```

**To Reproduce**
<!-- Please provide a relevant code snippets to reproduce this bug. -->
Just trigger search result with url including "SPACE"

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
space must be encoded as %20

**Please complete the following information:**
- LangChain4j version: 0.33.0
- LLM(s) used: openai
- Java version: 21
- Spring Boot version (if applicable): 

**Additional context**
<!-- Add any other context about the problem here. -->

@langchain4j It's so frequently triggered. (also stack trace contains example issue)

reproduction procedure.

1. Define ContentRetriever.
```
  WebSearchEngine webSearchEngine() {
    // https://app.tavily.com/sign-in
    return TavilyWebSearchEngine.builder().apiKey(<<API_KEY>>).build();
  }
  
  ContentRetriever webSearchContentRetriever(WebSearchEngine webSearchEngine) {
    return WebSearchContentRetriever.builder()
        .webSearchEngine(webSearchEngine)
        .maxResults(10)
        .build();
  }
 ```

2. Just query whatever (run below code 1 or 2 times may trigger meltdown)
```
webSearchContentRetriever.retrieve(Query.from("너희들이 게맛을 알어"));
```

```
Caused by: java.net.URISyntaxException: Illegal character in path at index 25: https://cboard.net/s/너희들이 게맛을 알어
        at java.base/java.net.URI$Parser.fail(URI.java:2995) ~[na:na]
        at java.base/java.net.URI$Parser.checkChars(URI.java:3166) ~[na:na]
        at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3248) ~[na:na]
        at java.base/java.net.URI$Parser.parse(URI.java:3196) ~[na:na]
        at java.base/java.net.URI.<init>(URI.java:645) ~[na:na]
        at java.base/java.net.URI.create(URI.java:930) ~[na:na]
        ... 31 common frames omitted
```

--- a/web-search-engines/langchain4j-web-search-engine-tavily/src/main/java/dev/langchain4j/web/search/tavily/TavilyWebSearchEngine.java
+++ b/web-search-engines/langchain4j-web-search-engine-tavily/src/main/java/dev/langchain4j/web/search/tavily/TavilyWebSearchEngine.java
@@ -98,7 +98,7 @@ public class TavilyWebSearchEngine implements WebSearchEngine {
 
     private static WebSearchOrganicResult toWebSearchOrganicResult(TavilySearchResult tavilySearchResult) {
         return WebSearchOrganicResult.from(tavilySearchResult.getTitle(),
-                URI.create(tavilySearchResult.getUrl()),
+                URI.create(tavilySearchResult.getUrl().replaceAll(" ", "%20")),
                 tavilySearchResult.getContent(),
                 tavilySearchResult.getRawContent(),
                 Collections.singletonMap("score", String.valueOf(tavilySearchResult.getScore())));

